src/Profilers/MonitorProfiler/Communication/CommandServer.cpp

```diff
@@ -94,70 +93,162 @@ void CommandServer::ListeningThread()
         if (FAILED(hr))
         {
             _logger->Log(LogLevel::Error, _LS("Failed to validate message: 0x%08x"), hr);
-            doEnqueueMessage = false;
+            *reinterpret_cast<HRESULT*>(response.Payload.data()) = hr;
+            SendMessage(client, response);
+            Shutdown(client);
+            continue;
         }
 
-        *reinterpret_cast<HRESULT*>(response.Payload.data()) = hr;
-
-        hr = client->Send(response);
-        if (FAILED(hr))
+        if (!IsControlCommand(message))
         {
-            _logger->Log(LogLevel::Error, _LS("Unexpected error when sending data: 0x%08x"), hr);
-            doEnqueueMessage = false;
+            ProcessMessage(message, client);
         }
-
-        hr = client->Shutdown();
-        if (FAILED(hr))
+        else
         {
-            _logger->Log(LogLevel::Warning, _LS("Unexpected error during shutdown: 0x%08x"), hr);
-            // Not fatal, keep processing the message
+            ProcessResetMessage(message, client);
         }
+    }
+}
+
+void CommandServer::ProcessMessage(const IpcMessage& message, std::shared_ptr<IpcCommClient> client)
+{
+    IpcMessage response;
+    response.CommandSet = static_cast<unsigned short>(CommandSet::ServerResponse);
+    response.Command = static_cast<unsigned short>(ServerResponseCommand::Status);
+    response.Payload.resize(sizeof(HRESULT));
+
+    *reinterpret_cast<HRESULT*>(response.Payload.data()) = S_OK;
+    SendMessage(client, response);
+    Shutdown(client);
+
+    CallbackInfo info;
+    info.Message = message;
+    bool unmanagedOnly = false;
+    if (SUCCEEDED(_unmanagedOnlyCallback(info.Message.CommandSet, unmanagedOnly)) && unmanagedOnly)
+    {
+        _unmanagedOnlyQueue.Enqueue(info);
+    }
+    else
+    {
+        _clientQueue.Enqueue(info);
+    }
+}
+
+void CommandServer::ProcessResetMessage(const IpcMessage& message, std::shared_ptr<IpcCommClient> client)
+{
+    IpcMessage response;
+    response.CommandSet = static_cast<unsigned short>(CommandSet::ServerResponse);
+    response.Command = static_cast<unsigned short>(ServerResponseCommand::Status);
+    response.Payload.resize(sizeof(HRESULT));
+
+    HRESULT hr = S_OK;
+
+    if ((message.CommandSet == static_cast<unsigned short>(CommandSet::StartupHook)) &&
+        (message.Command == static_cast<unsigned short>(StartupHookCommand::Stop) || message.Command == static_cast<unsigned short>(StartupHookCommand::ResetState)))
+    {
+        CallbackInfo managedCallbackInfo;
+        CallbackInfo nativeCallbackInfo;
+
+        CreateControlMessage(CommandSet::StartupHook, StartupHookCommand::Stop, managedCallbackInfo);
```

Comment:
Per Joe:
- Can we add a generic stop so that other future features can easily pick up that a reset happened